operator/cmd/root.go

```diff
@@ -365,17 +365,19 @@ func registerOperatorHooks(log *slog.Logger, lc cell.Lifecycle, llc *LeaderLifec
 	})
 }
 
-func initEnv(vp *viper.Viper) {
+func initEnv(logger *slog.Logger, vp *viper.Viper) {
 	// Prepopulate option.Config with options from CLI.
 	option.Config.SetupLogging(vp, binaryName)
-	option.Config.Populate(vp)
-	operatorOption.Config.Populate(vp)
+
+	option.Config.Populate(logger, vp)
 
 	// add hooks after setting up metrics in the option.Config
-	logging.DefaultLogger.Hooks.Add(metrics.NewLoggingHook())
+	logging.AddHooks(metrics.NewLoggingHook())
+
+	operatorOption.Config.Populate(vp)
```

Comment:
Shouldn't we keep the previous order:

1) populate global config
2) populate operator config
3) add logging hooks

Reply:
I've doubled checked after your comment but I can't find the reasons why. I've added comments to clarify the order. Let me know if it should be done differently.

Reply:
My main concern was about adding the logging hooks after starting the metrics in the operator's packages (though I don't know if there are any actual dependency with the metrics related options in operator config). But I see that you kept the order in the next iteration, all good then, thanks! :+1: